### PR TITLE
Add summary of changes for v2.15

### PR DESCRIPTION
**Pull request by janan07**
Describe your pull request here: This PR add the summary of changes for v2.15

List the file(s) included in this PR: v2_15_0.md

After creating the PR, follow the instructions in the comments.

### CODE REVIEW

File: docs/whats-new/release-notes/v2_15_0.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+### Zowe API Mediation Layer
+
+ -Displaying the base path in API Catalog has been fixed. ([#3297](https://github.com/zowe/api-layer/issues/3297))
+- Disabling EhCache using `apiml.caching.enabled=false` has been fixed. ([#3280](https://github.com/zowe/api-layer/issues/3280))
+
+### Zowe CLI
+
+#### Zowe CLI (Core)
+
+- Fixed race condition in `config convert-profiles` command that may fail to delete secure values for old profiles.
```
Comment (anaxceron) [inline docs/whats-new/release-notes/v2_15_0.md:96]: @adam-wolfe - What PR/issue link can we add to this item?

File: docs/whats-new/release-notes/v2_15_0.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+- Resolved technical currency by updating `socks` transitive dependency.
```
Comment (anaxceron) [inline docs/whats-new/release-notes/v2_15_0.md:98]: @adam-wolfe - What PR/issue link can we add to this item?

File: docs/whats-new/release-notes/v2_15_0.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+- Updated zos-files copy dsclp system tests to include large mock files.
```
Comment (anaxceron) [inline docs/whats-new/release-notes/v2_15_0.md:100]: @adam-wolfe - What PR/issue link can we add to this item? Also, would like to explain why we're doing this. Is it to improve the efficiency of the development process?

File: docs/whats-new/release-notes/v2_15_0.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+#### Zowe CLI Imperative Framework
+
+- Fixed race condition in `config convert-profiles` command that may fail to delete secure values for old profiles.
```
Comment (anaxceron) [inline docs/whats-new/release-notes/v2_15_0.md:104]: @adam-wolfe - What PR/issue link can we add to this item?

File: docs/whats-new/release-notes/v2_15_0.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+- Resolved issue in `ProfileInfo` where schema comparisons fail, specifically when comparing the cached schema against a command-based schema during registration.
```
Comment (anaxceron) [inline docs/whats-new/release-notes/v2_15_0.md:106]: @adam-wolfe - What PR/issue link can we add to this item?

File: docs/whats-new/release-notes/v2_15_0.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+- Resolved technical currency by updating `socks` transitive dependency.
```
Comment (anaxceron) [inline docs/whats-new/release-notes/v2_15_0.md:108]: @adam-wolfe - What PR/issue link can we add to this item?

File: docs/whats-new/release-notes/v2_15_0.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+
+- Updated `mustache` and `jsonschema` dependencies for technical currency.
```
Comment (anaxceron) [inline docs/whats-new/release-notes/v2_15_0.md:110]: @adam-wolfe - What PR/issue link can we add to this item?

File: docs/whats-new/release-notes/v2_15_0.md [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+
+- Fixed issue where secure property names could be returned for the wrong profile. ([Zowe Explorer #2633](https://github.com/zowe/vscode-extension-for-zowe/issues/2633))
+
+- Fixed issue when a property is not found in `ProfileInfo.updateProperty({forceUpdate: true})`. ([Zowe Explorer #2493](https://github.com/zowe/vscode-extension-for-zowe/issues/2493))
```
Comment (anaxceron) [inline docs/whats-new/release-notes/v2_15_0.md:114]: @adam-wolfe - Can we explain the benefit of fixing this bug?

File: docs/whats-new/release-notes/v2_15_0.md [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+
+#### MQ Plug-in for Zowe CLI
+
+#### IMS Plug-in for Zowe CLI
+
+ Added missing shrinkwrap
+
+#### z/OS FTP Plug-in for Zowe CLI
+
+- Fixed bug where upload dataset using Buffer, stead of string. ([2533](https://github.com/zowe/vscode-extension-for-zowe/issues/2533))
```
Comment (anaxceron) [inline docs/whats-new/release-notes/v2_15_0.md:128]: @adam-wolfe - I don't understand this note at all. Can you help explain what we did, and why we did it?

File: docs/whats-new/release-notes/v2_15_0.md [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+
+### Zowe API Mediation Layer
+
+- Timeout configuration values of the websocket webclient are now supported in the API Gateway. ([#3315](https://github.com/zowe/api-layer/issues/3315))
+
+### Zowe CLI
+
+#### Zowe CLI (Core)
+
+- Added `--binary` and `--encoding` options to the `zos-files edit` command in Zowe CLI V2.
```
Comment (anaxceron) [inline docs/whats-new/release-notes/v2_15_0.md:34]: @adam-wolfe - would like to explain the benefit of adding these two options.

File: docs/whats-new/release-notes/v2_15_0.md [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+
+- Hid the progress bar if the `CI` environment variable is set, or if the `FORCE_COLOR` environment variable is set to `0`. ([#1845](https://github.com/zowe/zowe-cli/issues/1845))
```
Comment (anaxceron) [inline docs/whats-new/release-notes/v2_15_0.md:36]: @adam-wolfe - would like to explain the benefit of adding these two options.

File: docs/whats-new/release-notes/v2_15_0.md [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+
+#### CICS Plug-in for Zowe CLI
+
+#### DB2 Plug-in for Zowe CLI
+
+#### MQ Plug-in for Zowe CLI
+
+#### IMS Plug-in for Zowe CLI
+
+- Added notices announcing the deprecation of the IMS Plug-in for Zowe CLI. ([#66](https://github.com/zowe/zowe-cli-ims-plugin/pull/66))
```
Comment (anaxceron) [inline docs/whats-new/release-notes/v2_15_0.md:52]: @adam-wolfe - can we explain _why_ we are deprecating the plug-in?

File: docs/whats-new/release-notes/v2_15_0.md [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+
+#### z/OS FTP Plug-in for Zowe CLI
+
+- Simplified the preparation for JCL system tests.
+- Updated the version of zos-node-accessor to 1.0.16.
```
Comment (anaxceron) [inline docs/whats-new/release-notes/v2_15_0.md:57]: @adam-wolfe - would like to add an issue/pr link to this

File: docs/whats-new/release-notes/v2_15_0.md [hunk 14]
```diff
@@ -0,0 +1,150 @@
...
+
+#### MQ Plug-in for Zowe CLI
+
+#### IMS Plug-in for Zowe CLI
+
+- Added notices announcing the deprecation of the IMS Plug-in for Zowe CLI. ([#66](https://github.com/zowe/zowe-cli-ims-plugin/pull/66))
+
+#### z/OS FTP Plug-in for Zowe CLI
+
+- Simplified the preparation for JCL system tests.
```
Comment (anaxceron) [inline docs/whats-new/release-notes/v2_15_0.md:56]: @adam-wolfe - would like to add an issue/pr link to this. would also like to explain the benefit of this change.

File: docs/whats-new/release-notes/v2_15_0.md (hunk 1 above)
Comment (adam-wolfe) [reply docs/whats-new/release-notes/v2_15_0.md:96]: https://github.com/zowe/zowe-cli/pull/2055 - I don't think we had an issue for this.

File: docs/whats-new/release-notes/v2_15_0.md (hunk 2 above)
Comment (adam-wolfe) [reply docs/whats-new/release-notes/v2_15_0.md:98]: https://github.com/zowe/zowe-cli/pull/2049

File: docs/whats-new/release-notes/v2_15_0.md (hunk 3 above)
Comment (adam-wolfe) [reply docs/whats-new/release-notes/v2_15_0.md:100]: https://github.com/zowe/zowe-cli/pull/2023 - This was done to improve system test coverage for edge cases involving large data sets.

File: docs/whats-new/release-notes/v2_15_0.md (hunk 10 above)
Comment (adam-wolfe) [reply docs/whats-new/release-notes/v2_15_0.md:34]: These options allow users to download and edit binary files and data sets or to edit files and data sets with a user-specified encoding.

Comment (adam-wolfe) [reply docs/whats-new/release-notes/v2_15_0.md:34]: Issue: https://github.com/zowe/zowe-cli/issues/1725

File: docs/whats-new/release-notes/v2_15_0.md (hunk 11 above)
Comment (adam-wolfe) [reply docs/whats-new/release-notes/v2_15_0.md:36]: In some cases, the progress bar might not display properly, which could make logs hard to read. Setting these environment variables gives users a way to hide the progress bar for environments where the progress bar does not work.

File: docs/whats-new/release-notes/v2_15_0.md (hunk 14 above)
Comment (adam-wolfe) [reply docs/whats-new/release-notes/v2_15_0.md:56]: https://github.com/zowe/zowe-cli-ftp-plugin/pull/151. It looks like the PR should reduce the amount of preparation needed to run system tests, but @zFernand0 might be able to give a good summary.

File: docs/whats-new/release-notes/v2_15_0.md (hunk 13 above)
Comment (adam-wolfe) [reply docs/whats-new/release-notes/v2_15_0.md:57]: https://github.com/zowe/zowe-cli-ftp-plugin/pull/149

File: docs/whats-new/release-notes/v2_15_0.md (hunk 5 above)
Comment (adam-wolfe) [reply docs/whats-new/release-notes/v2_15_0.md:106]: https://github.com/zowe/zowe-cli/pull/2045

File: docs/whats-new/release-notes/v2_15_0.md (hunk 6 above)
Comment (adam-wolfe) [reply docs/whats-new/release-notes/v2_15_0.md:108]: https://github.com/zowe/zowe-cli/pull/2049

File: docs/whats-new/release-notes/v2_15_0.md (hunk 7 above)
Comment (adam-wolfe) [reply docs/whats-new/release-notes/v2_15_0.md:110]: https://github.com/zowe/zowe-cli/pull/2024

File: docs/whats-new/release-notes/v2_15_0.md (hunk 8 above)
Comment (adam-wolfe) [reply docs/whats-new/release-notes/v2_15_0.md:114]: I guess the benefit is that extenders can set/update a property in a team config even if it is not defined in the schema without their application/extension failing. @zFernand0 may be able to give a better explanation.

File: docs/whats-new/release-notes/v2_15_0.md (hunk 9 above)
Comment (adam-wolfe) [reply docs/whats-new/release-notes/v2_15_0.md:128]: "stead" should be "instead". zos-node-accessor's uploadDataset function expects data in the form of a ReadableStream, Buffer, or path to a local file. 

Since the FTP plugin was passing a string rather than a Buffer, on Windows I believe there was an issue with JCL statements which contain "//" at the start of a line being misinterpreted as a local file path, causing the upload function to fail.

By uploading the data as a buffer, it resolves an issue with uploading JCL when using the plug-in on a computer running Windows.

@zFernand0 might have a more succinct way to explain it.

File: docs/whats-new/release-notes/v2_15_0.md [hunk 15]
```diff
@@ -0,0 +1,139 @@
...
+
+- Updated `mustache` and `jsonschema` dependencies for technical currency. ([Zowe CLI 2024](https://github.com/zowe/zowe-cli/pull/2024))
+
+- Fixed issue where secure property names could be returned for the wrong profile. ([Zowe Explorer #2633](https://github.com/zowe/vscode-extension-for-zowe/issues/2633))
+
+- Fixed issue when a property is not found in `ProfileInfo.updateProperty({forceUpdate: true})` so that extenders can set or update a property in a team configuration even if it is not defined in the schema without their application/extension failing. ([Zowe Explorer #2493](https://github.com/zowe/vscode-extension-for-zowe/issues/2493))
+
+#### IMS Plug-in for Zowe CLI
+
+- Added missing shrinkwrap.
```
Comment (anaxceron) [inline docs/whats-new/release-notes/v2_15_0.md:113]: @adam-wolfe -- sorry, one more: what would be the PR/issue link for this? and what does this magical shrinkwrap do in this instance?

Comment (adam-wolfe) [reply docs/whats-new/release-notes/v2_15_0.md:113]: https://github.com/zowe/zowe-cli-ims-plugin/pull/65 - the npm shrinkwrap is used to specify specific versions of dependencies that are installed with the plug-in.

File: docs/whats-new/release-notes/v2_15_0.md (hunk 8 above)
Comment (zFernand0) [reply docs/whats-new/release-notes/v2_15_0.md]: I don't think I could've said it better myself 😋 
That's exactly it! 🥳

File: docs/whats-new/release-notes/v2_15_0.md (hunk 9 above)
Comment (zFernand0) [reply docs/whats-new/release-notes/v2_15_0.md]: That explanation is perfect! 🙏🏽 

I think we can say something along these lines:
- Fixed a Windows specific issue where uploading a dataset with JCL content hangs preventing further actions through the zFTP VSCode Extension.

File: docs/whats-new/release-notes/v2_15_0.md (hunk 14 above)
Comment (zFernand0) [reply docs/whats-new/release-notes/v2_15_0.md]: This is something that can be omitted from the Zowe Docs site.
While it does help us deliver higher quality features faster by making the codebase more maintainable, this is not something that the user will be able to perceive (like a new command) with this new version of the plug-in.

**Comment by balhar-jakub** [comment]
Seems to be good for API ML.

File: docs/whats-new/release-notes/v2_15_0.md [hunk 16]
```diff
@@ -0,0 +1,138 @@
...
+
+#### Zowe Explorer FTP Extension
+
+- See the [Zowe Explorer FTP Extension](https://github.com/zowe/vscode-extension-for-zowe/blob/main/packages/zowe-explorer-ftp-extension/CHANGELOG.md) changelog for updates included in this release.
+
+#### Zowe Explorer ESLint Plug-in
+
+- See the [Zowe Explorer ESLint Plug-in](https://github.com/zowe/vscode-extension-for-zowe/blob/main/packages/eslint-plugin-zowe-explorer/CHANGELOG.md) changelog for updates included in this release.
+
+### Vulnerabilities fixed
```
Comment (anaxceron) [inline docs/whats-new/release-notes/v2_15_0.md:136]: @janan07 -- This content is missing. Wasn't sure if you were waiting to add this yourself later on, but wanted to point this out just in case.
